doc/design/OrionMongoSink.md

```diff
@@ -0,0 +1,109 @@
+
+###Example
+Assuming the following Flume event is created from a notified NGSI context data (the code below is an <i>object representation</i>, not any real data format):
+
+    flume-event={
+        headers={
+	        content-type=application/json,
+	        fiware-service=vehicles,
+	        fiware-servicepath=4wheels,
+	        timestamp=1429535775,
+	        transactionId=1429535775-308-0000000000,
+	        ttl=10,
+	        destination=car1_car
+        },
+        body={
+	        entityId=car1,
+	        entityType=car,
+	        attributes=[
+	            {
+	                attrName=speed,
+	                attrType=kmh,
+	                attrValue=112.9
+	            }
+	        ]
+	    }
+    }
+
+Then `OrionMongoSink` will persist the data within the body as:
+
+    $ mongo
+    MongoDB shell version: 2.6.9
+    connecting to: test
+    > show databases
+    admin              (empty)
+    local              0.031GB
+    vehicles           0.031GB
+    test               0.031GB
+    > use vehicles
+    switched to db vehicles
+    > show collections
+    4wheels
+    4wheels_car1_car
+    4wheels_car1_car_speed
+    system.indexes
+    > db.4wheels.fnd()
+    { "_id" : ObjectId("5534d143fa701f0be751db82"), "recvTime" : "2015-04-20T12:13:22.41.124UTC", "entityId" : "car1", "entityType" : "car", "attrName" : "speed", "attrType" : "kmh", "attrValue" : "112.9" }
+    > db.4wheels_car1_car.find()
+    { "_id" : ObjectId("5534d143fa701f0be751db82"), "recvTime" : "2015-04-20T12:13:22.41.412UTC", "attrName" : "speed", "attrType" : "kmh", "attrValue" : "112.9" }
+    > db.4wheels_car1_car_speed.find()
+    { "_id" : ObjectId("5534d143fa701f0be751db82"), "recvTime" : "2015-04-20T12:13:22.41.560UTC", "attrType" : "kmh", "attrValue" : "112.9" }
+
+NOTE: the results for the three different data models (<i>collection-per-service-path</i>, <i>collection-per-entity</i> and <i>collection-per-attribute</i>) are shown respectively; and no database prefix nor collection prefix was used (see next section for more details).
+
+##Configuration
+`OrionMongoSink` is configured through the following parameters:
+
+| Parameter | Mandatory | Default value | Comments |
+|---|---|---|---|
+| type | yes | N/A | com.telefonica.iot.cygnus.sinks.OrionMongoSink |
+| channel | yes | N/A |
+| mongo_host | no | localhost |
+| mongo_port | no | 27017 |
+| mongo_username | no | <i>empty</i> |
+| mongo_password | no | <i>empty</i> |
+| data_model | no | collection-per-entity | Under study
+| db_prefix | no | <i>empty</i> | Should not be public
+| collection_prefix | no | <i>empty</i> | Should not be public
+
+A configuration example could be:
+
+    cygnusagent.sinks = mongo-sink
+    cygnusagent.channels = mongo-channel
+    ...
+    cygnusagent.sinks.mongo-sink.type = com.telefonica.iot.cygnus.sinks.OrionMongoSink
+    cygnusagent.sinks.mongo-sink.channel = mongo-channel
+    cygnusagent.sinks.mongo-sink.mongo_uri = 130.206.80.34:27017
```

Comment:
Fixed in https://github.com/telefonicaid/fiware-cygnus/commit/40bcc9d7925a4f871aa05b6002783ce32ae3ffbd